test/Npgsql.Tests/Types/NumericTypeTests.cs

```diff
@@ -335,6 +335,32 @@ public void Numeric()
                         }
                     }
                 }
+
+                //NOTE: it is important to test rounding on both even and odd numbers to make sure
+                //midpoint rounding is AwayFromZero instead of ToEven https://msdn.microsoft.com/en-us/library/system.midpointrounding.aspx
+                //This is necessary to match psql.
+                var rounding = new Tuple<string, decimal>[] {
+                    new Tuple<string,decimal>("'1'::numeric(10,2)", 1.00M),     //test scale
```

Comment:
This test needs to be cleaned up and possibly split into several ones... It starts with a check on `1234567.890123` that is very similar to what you've added here (so the two need to be merged).

Instead of having an array of Tuple, it's better to have a separate test with TestCases (see https://github.com/npgsql/npgsql/blob/f3de4c2a9a24930bc3351b0ad0ab4cc5f792b26d/test/Npgsql.Tests/SqlQueryParserTests.cs#L53 for an example).

Also, in general the Tuple class is a little obsolete ever since C# 7 introduced value tuples with integrated syntax in C#.

Reply:
Good points, I can do that.

Reply:
nit: spaces after commas

Reply:
@YohDeadfall sure enough! Can hardly believe I missed that. At first I thought you meant there shouldn't be spaces after commas...

Reply:
@roji I have a little time today after all. Does the latest commit of tests seem acceptable?

Reply:
Is it same as `NumericScaleCasting`? If yes, remove it.